src/components/Market/OrderCard/components/Swap/Swap.tsx

```diff
@@ -165,8 +205,15 @@ const Swap: React.FC = () => {
   )
 
   const handleSetMax = useCallback(() => {
-    tokenBalance && onUserInput(tokenBalance)
-  }, [tokenBalance, onUserInput])
+    if (orderType === Operation.LONG && !parseEther(prem).isZero()) {
+      const maxOptions = parseEther(tokenBalance)
```

Comment:
token balance here should be underlying amount, but its the long option amount since its changed a few lines above